src/lib/utils/time.ts

```diff
@@ -0,0 +1,89 @@
+export const millisecondsPerSecond = 1000
+export const secondsPerMinute = 60
+export const minutesPerHour = 60
+export const hoursPerDay = 24
+export const daysPerWeek = 7
+
+export const millisecondsPerMinute = millisecondsPerSecond * secondsPerMinute
+export const millisecondsPerHour = millisecondsPerMinute * minutesPerHour
+export const millisecondsPerDay = millisecondsPerHour * hoursPerDay
+export const millisecondsPerWeek = millisecondsPerDay * daysPerWeek
+export const previousSevenDays = millisecondsPerWeek
+export const lastThirtyDays = millisecondsPerDay * 30
+
+export const currentYear = new Date().getFullYear()
+
+export const millisecondsSince = (date: Date) => {
+  const now = new Date()
+  return date > now ? 0 : now.getTime() - date.getTime()
+}
+
+export const timeSince = (date: Date) => {
+  const timeDifference = millisecondsSince(date)
+  while (timeDifference != 0) {
```

Comment:
This isn't necessary, I think it's okay to return `"0s"` if timeDifference is 0. Possibly you could just return an empty string.